src/main/java/net/ckozak/ExecutorBenchmarks.java

```diff
@@ -124,12 +140,58 @@ public void tearDown() throws InterruptedException {
         }
     }
 
+    @State(Scope.Thread)
+    public static class ExecutingTasks implements Runnable {
+
+        Blackhole blackhole;
+        Future<?>[] pendingTasks;
+        int burstSize;
+
+        @Setup(Level.Iteration)
+        public void init(Blackhole blackhole, ExecutorBenchmarks benchmarks) {
+            burstSize = benchmarks.burstSize;
+            pendingTasks = new FutureTask<?>[benchmarks.burstSize];
+            this.blackhole = blackhole;
+        }
+
+        @Override
+        public void run()
+        {
+            if (DELAY_CONSUMER > 0) {
+                Blackhole.consumeCPU(DELAY_CONSUMER);
+            }
+            // not sure this one is needed
+            blackhole.consume(Thread.currentThread().getId());
+        }
+
+        public void submitBurst(ExecutorService executor)
+        {
+            Future<?>[] pendingTasks = this.pendingTasks;
+            for (int i = 0, size = burstSize; i < size; i++)
+            {
+                pendingTasks[i] = executor.submit(this);
```

Comment:
I wonder if we should use a single CountDownLatch instead of creating N FutureTasks to reduce the per-task allocation overhead?

Reply:
latch performance characteristics are undesirable here

Reply:
Yes about reducing the allocation pressure.

But I have preserved using Future::get, I can send a follow up or if we are not interested into using it :) thanks for the review

Reply:
Up to you, if you think it's a helpful benchmark I'd be happy to maintain it as an additional option here. Ideally I'd transfer this code to an official location.